### ISSUE TITLE
Modrinth Game Versions don't show ranges

### ISSUE THREAD

**Issue by n-aspen** [badges/shields#10346]
### Are you experiencing an issue with...

shields.io

### 🐞 Description

When the latest release of a Modrinth project has a range of supported version, each version is listed individually instead of as a range as it is on Modrinth. This causes *very long* badges in some cases. 

### 🔗 Link to the badge

![](https://img.shields.io/modrinth/game-versions/P1OZGk5p)
![image](https://github.com/badges/shields/assets/16448318/17b0fbc4-8890-457f-a0e8-cf0c418c4eb3)
```https://img.shields.io/modrinth/game-versions/P1OZGk5p```
https://modrinth.com/plugin/viaversion

### 💡 Possible Solution

Using the same range behavior when multiple versions are supported as Modrinth would be ideal and is what I was expecting.

**Comment by PyvesB** [comment]
Hello @n-aspen 👋🏻 

Thanks for reporting, that badge is indeed very long. 😄 

[The API](https://docs.modrinth.com/#tag/versions) doesn't seem to be returning any range info, just a list of versions. If we get `["1.17", "1.18", "1.18.1", "1.18.2", "1.19"]` back, one may be tempted to conclude that the range is `1.17-1.19`, but that's technically not quite correct as version 1.17.1 is missing from the list. I'm wondering whether the Modrinth website handles these cases correctly.

A less elegant but more robust solution would be to put an ellipsis in the middle, e.g. `1.17 | 1.18 | ... | 1.19 | 1.19.1` when there are for example five or more versions returned by the API.

**Comment by n-aspen** [comment]
That's probably a reasonable solution for 90% of use cases tbh. Most of the time a project will support a lower and upper bound for versions. 

I wanted to see how Modrinth handled it as their website is open source, and I think [this](https://github.com/modrinth/knossos/blob/master/helpers/projects.js) is it? I'm not completely certain though, and best as I can tell it'd be way out of scope for this.

**Comment by n-aspen** [comment]
I found a definitive answer to this question 
> I'm wondering whether the Modrinth website handles these cases correctly.

This plugin [1u6JkXh5](https://modrinth.com/plugin/worldedit), has an [old-ish release](https://modrinth.com/plugin/worldedit/version/DfjpjdP5) that only has `1.17.1, 1.18.2, 1.19.4-1.20.4` marked as supported, which matches this list in [the API](https://api.modrinth.com/v2/project/1u6JkXh5/version/DfjpjdP5), where it's only made a range out of the `1.20.x` entries because they're listed individually. 
```
0	"1.17.1"
1	"1.18.2"
2	"1.19.4"
3	"1.20"
4	"1.20.1"
5	"1.20.2"
6	"1.20.3"
7	"1.20.4"
```
![image](https://github.com/badges/shields/assets/16448318/435c30f6-bccb-427a-add6-b1bb860bdf36)

**Comment by PyvesB** [comment]
Thanks for digging that out! I'd foresee two issues with following a similar approach:
* if the ranges are broken down in too many parts due to unsupported game versions here and there, the badge would still risk being too long.
* we'd need to keep track of all known game versions, and make sure the listing is updated on a regular basis.

I'll let other maintainers weigh in, but I'd personally be in favour of a simpler and more robust solution, e.g. something ellipsis-based like `1.17 | 1.18 | ... | 1.19 | 1.19.1`.

**Comment by chris48s** [comment]
Ellipsis is definitely the easiest thing to do, and it puts a cap on the length. There are going to be some situations where it is not that helpful

There is a Modrinth API endpoint which gives all the available game versions. https://api.modrinth.com/v2/tag/game_version

```
curl "https://api.modrinth.com/v2/tag/game_version" | jq '[.[] | select(.version_type == "release") | .version]'
```

We could potentially use that to work out if a list of supported game versions on a project can be displayed as a range. It would be an extra API request, but we could do them in parallel or cache the `game_version` call.
